agent/lib_php_amqplib.c

```diff
@@ -0,0 +1,770 @@
+ * the existence of the php-amqplib is a known quantity so calling the following
+ * lambda will result in the PhpAmqpLib\\Package class being loaded.
+ */
+void nr_php_amqplib_handle_version() {
+  char* version = NULL;
+  zval retval_zpd;
+  int result = FAILURE;
+
+  result = zend_eval_string(
+      "(function() {"
+      "     $nr_php_amqplib_version = '';"
+      "     try {"
+      "          $nr_php_amqplib_version = PhpAmqpLib\\Package::VERSION;"
+      "     } catch (Throwable $e) {"
+      "     }"
+      "     return $nr_php_amqplib_version;"
+      "})();",
+      &retval_zpd, "Get nr_php_amqplib_version");
+
+  /* See if we got a non-empty/non-null string for version. */
+  if (SUCCESS == result) {
+    if (nr_php_is_zval_non_empty_string(&retval_zpd)) {
+      version = Z_STRVAL(retval_zpd);
+    }
+  }
+
+  if (NRINI(vulnerability_management_package_detection_enabled)) {
+    /* Add php package to transaction */
+    nr_txn_add_php_package(NRPRG(txn), PHP_PACKAGE_NAME, version);
+  }
+
+  nr_txn_suggest_package_supportability_metric(NRPRG(txn), PHP_PACKAGE_NAME,
+                                               version);
+
+  zval_dtor(&retval_zpd);
+}
+
+/*
+ * Purpose : Retrieves host and port from an AMQP Connection and sets the
+ * host/port values in the message_params.
+ *
+ * Params  : 1. PhpAmqpLib\Connection family of connections that inherit from
+ * AbstractConnection
+ *           2. nr_segment_message_params_t* message_params that will be
+ * modified with port and host info, if available
+ *
+ * Returns : None
+ *
+ * See here for more information about the AbstractConnection class that all
+ * Connection classes inherit from:
+ * https://github.com/php-amqplib/php-amqplib/blob/master/PhpAmqpLib/Connection/AbstractConnection.php
+ */
+static inline void nr_php_amqplib_get_host_and_port(
+    zval* amqp_connection,
+    nr_segment_message_params_t* message_params) {
+  zval* amqp_server = NULL;
+  zval* amqp_port = NULL;
+  zval* connect_constructor_params = NULL;
+
+  if (NULL == amqp_connection || NULL == message_params) {
+    return;
+  }
+
+  /* construct params are always saved to use for cloning purposes. */
+
+  if (!nr_php_is_zval_valid_object(amqp_connection)) {
+    return;
+  }
+
+  connect_constructor_params
+      = nr_php_get_zval_object_property(amqp_connection, "construct_params");
+  if (nr_php_is_zval_valid_array(connect_constructor_params)) {
+    return;
+  }
+
+  amqp_server
+      = nr_php_zend_hash_index_find(Z_ARRVAL_P(connect_constructor_params),
+                                    AMQP_CONSTRUCT_PARAMS_SERVER_INDEX);
+  if (nr_php_is_zval_non_empty_string(amqp_server)) {
+    message_params->server_address
+        = ENSURE_PERSISTENCE(Z_STRVAL_P(amqp_server));
+  }
+
+  amqp_port = nr_php_zend_hash_index_find(
+      Z_ARRVAL_P(connect_constructor_params), AMQP_CONSTRUCT_PARAMS_PORT_INDEX);
+  if (IS_LONG != Z_TYPE_P((amqp_port))) {
```

Comment:
`nr_php_is_zval_valid_integer` is what is needed here (it also checks for NULL):
```suggestion
  if (nr_php_is_zval_valid_integer(amqp_port)) {
```

Reply:
It said outdated suggestions cannot be applied.
suggested applied here:
[90da3b7](https://github.com/newrelic/newrelic-php-agent/pull/1009/commits/90da3b7c75ffb66a1d1548f89231d44f36edf022)